packages/infra/lib/ihe-stack.ts

```diff
@@ -13,89 +13,96 @@ import { LambdaLayers } from "./shared/lambda-layers";
 interface IHEStackProps extends StackProps {
   config: EnvConfig;
   vpc: ec2.IVpc;
-  alarmAction: SnsAction | undefined;
   lambdaLayers: LambdaLayers;
-  publicZone: r53.IHostedZone;
+  alarmAction: SnsAction | undefined;
 }
 
-export function createIHEStack(stack: Construct, props: IHEStackProps) {
-  //-------------------------------------------
-  // API Gateway
-  //-------------------------------------------
-  if (!props.config.iheGateway?.subdomain) {
-    throw new Error("Must define subdomainmain if building the IHE stack!");
-  }
+export class IHEStack extends Stack {
+  constructor(scope: Construct, id: string, props: IHEStackProps) {
+    super(scope, id, props);
+    //-------------------------------------------
+    // API Gateway
+    //-------------------------------------------
+    if (!props.config.ihe?.subdomain) {
+      throw new Error("Must define subdomainmain if building the IHE stack!");
+    }
 
-  if (!props.config.iheGateway?.certArn) {
-    throw new Error("Must define cert arn if building the IHE stack!");
-  }
+    if (!props.config.ihe?.gatewayCertArn) {
+      throw new Error("Must define cert arn if building the IHE stack!");
+    }
```

Comment:
We don't need to check the IHE Gateway properties, as they are required, just need to check whether the IHE Gateway is present or not.